Dart/src/com/jetbrains/lang/dart/ide/generation/CreateNamedConstructorFix.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2000-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jetbrains.lang.dart.ide.generation;
+
+import com.intellij.codeInsight.template.Template;
+import com.intellij.codeInsight.template.TemplateManager;
+import com.intellij.codeInsight.template.impl.MacroCallNode;
+import com.intellij.codeInsight.template.macro.CompleteSmartMacro;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.jetbrains.lang.dart.psi.DartClass;
+import com.jetbrains.lang.dart.psi.DartComponent;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Iterator;
+import java.util.Set;
+
+public class CreateNamedConstructorFix extends BaseCreateMethodsFix<DartComponent> {
+  public CreateNamedConstructorFix(DartClass dartClass) {
+    super(dartClass);
+  }
+
+  @Override
+  protected void processElements(@NotNull Project project, @NotNull Editor editor, Set<DartComponent> elementsToProcess) {
+    final TemplateManager templateManager = TemplateManager.getInstance(project);
+    anchor = doAddMethodsForOne(editor, templateManager, buildFunctionsText(templateManager, elementsToProcess), anchor);
+  }
+
+  @Override
+  @NotNull
+  protected String getNothingFoundMessage() {
+    return ""; // can't be called actually because processElements() is overridden
+  }
+
+  protected Template buildFunctionsText(TemplateManager templateManager, Set<DartComponent> elementsToProcess) {
+    final Template template = templateManager.createTemplate(getClass().getName(), DART_TEMPLATE_GROUP);
+    template.setToReformat(true);
+
+    //noinspection ConstantConditions
+    template.addTextSegment(myDartClass.getName());
+    template.addTextSegment(".");
+    template.addVariable(new MacroCallNode(new CompleteSmartMacro()), true);
```

Comment:
CompleteSmartMacro does nothing but showing 'No suggestions' because we do not have smart completion in Dart. Standard completion won't help either as it doesn't suggest anything in this case. I suggest to add some default name, e.g. 'name' or 'constructor' and simply stop on it.
```
template.addVariable(new TextExpression("constructor"), true);
```
It becomes preselected, so user can start typing desired name.
![image](https://cloud.githubusercontent.com/assets/4499789/12359215/41cc79b8-bbc2-11e5-8d19-069f58869cb4.png)